packages/devtools_app/lib/src/inspector/inspector_tree.dart

```diff
@@ -176,129 +172,24 @@ abstract class InspectorTreeNode {
-              final int blue = JsonUtils.getIntMember(properties, 'blue');
-              String radix(int chan) => chan.toRadixString(16).padLeft(2, '0');
-              if (alpha == 255) {
-                description = '#${radix(red)}${radix(green)}${radix(blue)}';
-              } else {
-                description =
-                    '#${radix(alpha)}${radix(red)}${radix(green)}${radix(blue)}';
-              }
-
-              final Color color = Color.fromARGB(alpha, red, green, blue);
-              builder.addIcon(_colorIconMaker.getCustomIcon(color));
-              break;
-            }
-
-          case 'IconData':
-            {
-              final int codePoint =
-                  JsonUtils.getIntMember(properties, 'codePoint');
-              if (codePoint > 0) {
-                final DevToolsIcon icon =
-                    FlutterMaterialIcons.getIconForCodePoint(codePoint);
-                if (icon != null) {
-                  builder.addIcon(icon);
-                }
-              }
-              break;
-            }
-        }
-      }
-
-      if (_showRenderObjectPropertiesAsLinks &&
-          propertyType == 'RenderObject') {
-        textStyle = textStyle..merge(inspector_text_styles.link);
-      }
-
-      // TODO(jacobr): custom display for units, iterables, and padding.
-      _renderDescription(builder, description, textStyle, isProperty: true);
-
-      if (diagnostic.level == DiagnosticLevel.fine &&
-          diagnostic.hasDefaultValue) {
-        builder.appendText(' ', textStyle);
-        builder.addIcon(defaultIcon);
+  /// Returns whether the node is currently visible in the tree.
+  void updateShouldShow(bool value) {
+    if (value != _shouldShow) {
+      _shouldShow = value;
+      for (var child in children) {
+        child.updateShouldShow(value);
       }
-    } else {
-      // Non property, regular node case.
-      if (name?.isNotEmpty == true && diagnostic.showName && name != 'child') {
-        if (name.startsWith('child ')) {
-          builder.appendText(name, inspector_text_styles.unimportant);
-        } else {
-          builder.appendText(name, textStyle);
-        }
-
-        if (diagnostic.showSeparator) {
-          builder.appendText(
-              diagnostic.separator, inspector_text_styles.unimportant);
-          if (diagnostic.separator != ' ' &&
-              diagnostic.description.isNotEmpty) {
-            builder.appendText(' ', inspector_text_styles.unimportant);
-          }
-        }
-      }
-
-      if (!diagnostic.isSummaryTree && diagnostic.isCreatedByLocalProject) {
-        textStyle = textStyle.merge(inspector_text_styles.regularBold);
-      }
-
-      _renderDescription(builder, diagnostic.description, textStyle,
-          isProperty: false);
     }
-    _renderObject = builder.build();
-    return _renderObject;
   }
 
-  InspectorTreeNodeRender _renderObject;
+  bool _shouldShow;
+  bool shouldShow() {
```

Comment:
can this be a getter instead?

Reply:
renamed to a getter even though it is a method in the similar mixin as this method is cached and efficient vs O(level).